### PR TITLE
Fix static analysis tools

### PR DESCRIPTION
**Pull request by Pahandrovich**

### CODE REVIEW

**Comment by napetrov** [comment]
New  code introduce 3 code factor issues - suggest to fix them now . Overall looks good

File: daal4py/sklearn/decomposition/_pca_0_23.py [hunk 1]
```diff
@@ -486,7 +486,7 @@ def transform(self, X):
             if self.mean_ is not None:
                 X = X - self.mean_
                 X_transformed = np.dot(X, self.components_.T)
```
Comment (PetrovKP) [inline daal4py/sklearn/decomposition/_pca_0_23.py:488]: This is a mistake. It should be like this:
```python
        if self.mean_ is not None:
            X = X - self.mean_
        X_transformed = np.dot(X, self.components_.T)
        if self.whiten:
            X_transformed /= np.sqrt(self.explained_variance_)
        return X_transformed
```
Now there is a chance that you will return None. This is bad.

File: generator/format.py [hunk 2]
```diff
@@ -150,7 +150,7 @@ def __init__(self, name, typ, const, dflt, inpt, algo, doc):
                     pd = (pydefaults[typ] if dflt == True else dflt).rsplit('::', 1)[-1]
                     sphinx_default = '"{}"'.format(pd) if typ == 'std::string' else '{}'.format(pd)
                     pydefault = ' = {}'.format(sphinx_default)
-                    cppdefault = ' = {}'.format(cppdefaults[typ] if dflt == True else dflt) if dflt != None else ''
+                    cppdefault = ' = {}'.format(cppdefaults[typ] if dflt == True else dflt)
```
Comment (PetrovKP) [inline generator/format.py:153]:
```suggestion
                    cppdefault = ' = {}'.format(cppdefaults[typ] if dflt else dflt)
```

File: generator/gen_daal4py.py [hunk 3]
```diff
@@ -679,7 +679,7 @@ def prepare_hlwrapper(self, ns, mode, func, no_dist, no_stream):
 
             for p in all_params:
                 pns, tmp = splitns(p)
-                if not tmp.startswith('_') and not ignored(pns, tmp):
+                if tmp != None and not tmp.startswith('_') and not ignored(pns, tmp):
```
Comment (PetrovKP) [inline generator/gen_daal4py.py:682]:
```suggestion
                if tmp is not None and not tmp.startswith('_') and not ignored(pns, tmp):
```

**Review by PetrovKP — CHANGES_REQUESTED** [review]

**Review by PetrovKP — APPROVED** [review]

**Review by napetrov — APPROVED** [review]

**Pull request merged**